independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeDeploymentManager.java

```diff
@@ -104,17 +104,29 @@ public BeanFactory.BeanInstance<?> apply(Class<?> aClass) {
                         return info.getFactoryCreator().apply(aClass).createInstance();
                     }
                 });
+
+        // sanitise the prefix for our usage to make it either an empty string, or something which starts with a / and does not
+        // end with one
+        String prefix = rootPath;
+        if (prefix != null) {
+            prefix = sanitizePathPrefix(prefix);
+        } else {
+            prefix = "";
+        }
+        if ((applicationPath != null) && !applicationPath.isEmpty()) {
+            prefix = prefix + sanitizePathPrefix(applicationPath);
+        }
```

Comment:
Why is this needed?

Reply:
I've just moved this piece of code a bit further up, because I need the prefix to compute the full resource paths, including the `quarkus.http.root-path` if it's defined.

Reply:
Ah, thanks!